integration/testrunner/grpc.go

```diff
@@ -170,13 +193,24 @@ func testGRPCEndpoint(ctx context.Context, t *testing.T, g *GRPC) {
 			}
 			t.Logf("Received value %v", place.Value)
 		}
-		query := g.makeQueryForGRPCStream(id)
-		queryIfAuditLogsExistWithRetries(ctx, t, query, g.Config, "server_stream_fibonacci", int64(places))
+		query := g.makeQueryForGRPC(id)
+		t.Log(query.Q)
+		results := queryIfAuditLogsExistsWithRetries(ctx, t, query, g.Config, "server_stream_fibonacci")
+		wantNum := places
+		if len(results) != wantNum {
+			t.Errorf("log number doesn't match (-want +got):\n - %d\n + %d\n", wantNum, len(results))
+		} else {
```

Comment:
Do we need such else here? I thought we don't need else here

Reply:
Same above

Reply:
Same above, If len(results) != wantNum, we should not keep comparing the logentry, we can just exit with return. But I don't have strong perference here. I just personally don't quite like if else block

Reply:
My idea here is that even the number doesn't match, it would still be good to check if fields matches, which would provide more context for debug when tests fail.

Reply:
In your current logic, you still won't check field matches when the number doesn't match

Reply:
Fixed.